### ISSUE TITLE
Joi 1.16 update : Error: requires a joi schema object

### ISSUE THREAD

**Issue by HugoHeneault** [lightsofapollo/joi-to-json-schema#57]
While trying to convert my joi schema to JSON I got an error : `AssertionError [ERR_ASSERTION]: requires a joi schema object`

Here is my code :

```
import * as Joi from "@hapi/joi";

const convert = require("joi-to-json-schema");

const  report = Joi.object({
      part: Joi.string()
          .valid(['a', 'b', 'c'])
          .required(),
     })

convert(report);
```

This seems to be related to https://github.com/hapijs/joi/issues/1834

We should use `Joi.isSchema()` instead.

I will try to provide a PR using isSchema when available then fallback to isJoi if it's OK for you.

**Comment by HugoHeneault** [comment]
So looking deeper to this issue, I figured out that, as said on  [joi v16 release notes](https://github.com/hapijs/joi/issues/2037):
> joi v16.0.0 is a massive release with many changes to every corner of the module and its extensions. The entire code base has been rewritten with many new features and core components.

Updating `joi-to-json-schema` to work with this new `joi` version will be more than just using `Joi.isSchema()`. I'm not sure I'll have enough time to go through all the fixes so I might downgrade my joi version for now. 

If anyone need some help for doing the upgrade I could give a hand but not carry all by myself 😦

**Comment by yeongjet** [comment]
Hope this project support the joi v16.

**Comment by yeongjet** [comment]
I forked this project in my [repo](https://github.com/yeongjet/joi-to-json-schema) then rewrote some code using typescript and made some adjustments to fit joi v16.0, almost test cases passed. May I make a pull request? @lightsofapollo

**Comment by BrianSo** [comment]
@yeongjet I tried your repo and it is working, I really hope you could contribute your work to the repo.

But I found that you're changing all files to typescript and the commit message is `update`. Could you explain whether you have done any thing other than setting up a typescript development environment in the first `update` commit?

And did you remove any previous test cases?

**Comment by yeongjet** [comment]
@BrianSo No, in order to make these test cases running in jest, I have to make syntax modification, there's no difference in fundamental. Except these, the most noteworthy in my commit is remapping the fields and make them works in original logic.

**Comment by lightsofapollo** [comment]
@yeongjet Yes! This repo is in maintenance mode but I'd love some contributions particularly in typescript/new joi direction.

**Comment by yeongjet** [comment]
@lightsofapollo There has a lot of changes in my commit,  which branch should be merged?  maybe create a new branch?

**Comment by simlu** [comment]
We'd love to see joi 1.16 supported as well

**Comment by simlu** [comment]
Any update on this?

**Comment by vgjenks** [comment]
@lightsofapollo @yeongjet - any chance this will get merged? This is a package I really need, and it's completely broken in latest Joi. Looks like no movement in over a year. :( Dead project?

**Comment by simlu** [comment]
@vgjenks We now use https://www.npmjs.com/package/joi-to-json - works really well for our use case

Edit: Here is the corresponding github repo https://github.com/kenspirit/joi-to-json

**Comment by vgjenks** [comment]
@simlu Works brilliantly, thank you! Will come back w/ feedback as I throw more complex schemas at it.
